projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -56,16 +57,23 @@ import { MultiSelectJustify } from './multi-select-justify';
         </ht-popover-trigger>
         <ht-popover-content>
           <div class="multi-select-content" [ngStyle]="{ 'min-width.px': triggerContainer.offsetWidth }">
+            <ng-container *ngIf="this.enableSearch">
+              <ht-search-box
+                class="search-bar"
+                (valueChange)="this.searchOptions($event)"
+                displayMode="${SearchBoxDisplayMode.NoBorder}"
+              ></ht-search-box>
+            </ng-container>
             <ng-container *ngIf="this.showAllOptionControl">
               <div class="multi-select-option all-options" (click)="this.onAllSelectionChange()">
                 <input class="checkbox" type="checkbox" [checked]="this.areAllOptionsSelected()" />
                 <span class="label">Select All</span>
               </div>
-
-              <ht-divider></ht-divider>
             </ng-container>
 
-            <div *ngFor="let item of items" (click)="this.onSelectionChange(item)" class="multi-select-option">
+            <ht-divider *ngIf="this.showAllOptionControl || this.enableSearch"></ht-divider>
```

Comment:
The new mocks supports "Select All" And "Clear Selected" as buttons (instead of it as an option). Is this something you are taking care of too?

Reply:
Not doing it at the moment. Currently doing things that would unblock me. Can circle back to it in the end.